### PR TITLE
[RET-279] Adding Btn Group Component

### PR DESCRIPTION
**Pull request by jcmagsay**
### What's this PR do?
- Add Button Group Component
- Add Button Group Option Component

### Does this add new dependencies? Is there an installation procedure? (bundle, bower, npm, etc)
N/A

### What are the relevant tickets?
https://mondorobot.atlassian.net/browse/RET-279

### Who should review this PR?
@jcmagsay @mhfen

### CODE REVIEW

File: src/client/components/button/button-group-option.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  render() {
+    const { isActive, buttonTxt, theme } = this.props;
+
+    return (
+      <Button
+        active={isActive}
+        onClick={this.handleClick}
+        theme={theme}
+      >
```
Comment (mhfen) [inline src/client/components/button/button-group-option.jsx]: How do you feel about this being a `label` instead of a `button`?

Comment (jcmagsay) [inline src/client/components/button/button-group-option.jsx]: I'm not against this suggestion at all, in fact it crossed my mind while developing. If we did make this a label, would you agree that extending the `.btn` class would be sufficient so we don't have to copy/paste the styles? Alternatively, we could remove the input and just use text, I only but the radios in for accessibility.

Comment (mhfen) [inline src/client/components/button/button-group-option.jsx]: Sounds good

**Comment by jcmagsay** [comment]
I updated the button group component to radio group to make things more semantic/expressive.

**Pull request merged**